### PR TITLE
WIP: VZ-6691: Deploy opensearch dashboards after ES is green

### PR DESCRIPTION
**Pull request by mgianatagh**
Defer the deployment of the OpenSearch dashboards until after ES is green.  This is to resolve the issue of a fresh install failing due to indexes being migrated.

### CODE REVIEW

**Review by anders-swanson — DISMISSED** [review]

File: pkg/resources/deployments/deployment_test.go [hunk 1]
```diff
@@ -54,8 +54,7 @@ func TestVMOFullDeploymentSize(t *testing.T) {
 		t.Error(err)
 	}
 	deployments := expected.Deployments
-	deployments = append(deployments, NewOpenSearchDashboardsDeployment(vmo))
-	assert.Equal(t, 5, len(deployments), "Length of generated deployments")
+	assert.Equal(t, 3, len(deployments), "Length of generated deployments")
```
Comment (anders-swanson) [inline pkg/resources/deployments/deployment_test.go:57]: why did deployments go down from 5 to 3 if only one deployment was removed?

File: pkg/resources/statefulsets/statefulset.go [hunk 2]
```diff
@@ -320,3 +320,44 @@ func createStatefulSetElement(vmo *vmcontrollerv1.VerrazzanoMonitoringInstance,
 		},
 	}
 }
+
+func NewOpenSearchDashboardsStatefulSet(vmo *vmcontrollerv1.VerrazzanoMonitoringInstance) *appsv1.StatefulSet {
+	var statefulSet *appsv1.StatefulSet
+
+	// Headless service for OpenSearch Dashboards
+	headlessService := resources.GetMetaName(vmo.Name, config.Kibana.Name)
```
Comment (anders-swanson) [inline pkg/resources/statefulsets/statefulset.go:328]: Is there a headless service for OSD?

File: pkg/resources/statefulsets/statefulset.go [hunk 3]
```diff
@@ -320,3 +320,44 @@ func createStatefulSetElement(vmo *vmcontrollerv1.VerrazzanoMonitoringInstance,
...
+	statefulSetName := resources.GetMetaName(vmo.Name, config.Kibana.Name)
+
+	if vmo.Spec.Kibana.Enabled {
+		elasticsearchURL := fmt.Sprintf("http://%s%s-%s:%d/", constants.VMOServiceNamePrefix, vmo.Name, config.ElasticsearchIngest.Name, config.ElasticsearchIngest.Port)
```
Comment (anders-swanson) [inline pkg/resources/statefulsets/statefulset.go:332]: Can you rename these vars as appropriate to opensearch, opensearchDashboards, etc.

File: pkg/resources/deployments/deployment.go [hunk 4]
```diff
@@ -252,41 +252,6 @@ func New(vmo *vmcontrollerv1.VerrazzanoMonitoringInstance, kubeclientset kuberne
...
 }
 
-func NewOpenSearchDashboardsDeployment(vmo *vmcontrollerv1.VerrazzanoMonitoringInstance) *appsv1.Deployment {
-	var deployment *appsv1.Deployment
-	if vmo.Spec.Kibana.Enabled {
-		elasticsearchURL := fmt.Sprintf("http://%s%s-%s:%d/", constants.VMOServiceNamePrefix, vmo.Name, config.ElasticsearchIngest.Name, config.ElasticsearchIngest.Port)
-		deployment = createDeploymentElement(vmo, nil, &vmo.Spec.Kibana.Resources, config.Kibana, config.Kibana.Name)
-
-		deployment.Spec.Strategy = appsv1.DeploymentStrategy{
-			Type: appsv1.RecreateDeploymentStrategyType,
```
Comment (anders-swanson) [inline pkg/resources/deployments/deployment.go:262]: STS cannot have recreate strategy?

Comment (mgianatagh) [reply pkg/resources/deployments/deployment.go:262]: For the STS I am setting the strategy to appsv1.RollingUpdateStatefulSetStrategyType.  The only other strategy type is OnDeleteStatefulSetStrategyType

File: pkg/resources/statefulsets/statefulset.go (hunk 3 above)
Comment (mgianatagh) [reply pkg/resources/statefulsets/statefulset.go:332]: Yes, I will do some renaming.  Just local vars?  Or should I also do some renaming of global vars like ElasticsearchMaster etc. in the config pkg?

**Pull request closed without merging**